2018/002-update-channels.md

```diff
@@ -0,0 +1,230 @@
+
+* update channel: normal maintenance updates, which are safe and non-intrusive
+* new version channel: version updates which can contain intrusive updates
+
+This RFC proposes:
+
+* a front-end change in the velum UI to enable the admin to switch to a new version
+channel
+* a back-end change in salt to perform the channel switch on all cluster nodes
+
+### Proposed change (Detailed)
+
+#### Tools/Services
+
+We need a set of tools/services to indicate, whenever a maintenance update or a new version
+update is ready to be installed. They should run on each worker node to be more independent
+and decoupled from velum. If velum would be the initiator of the update check, it might lead
+to inconsistent and long execution runs in larger clusters.
+
+##### "check for updates" service on each worker
+
+The current channel will offer maintenance updates, which need to be announced.
+
+The service:
+
+* runs once per day e.g.
+* is generally decoupled from velum, runs independently, and just reports.
+* reports availability of maintenance updates to velum via grains.
+
+##### "new version" checker service on each worker
+
+The next channel will offer possibly disruptive updates, which can be either features or
+version updates of certain components of the whole stack, which need to be announced.
+
+The service:
+
+* runs once per week or longer timeframe.
+* is generally decoupled from velum, runs independently, just reports.
+* checks for new versions (migration targets):
+  - query SCC API if there is a new compatible version available for this CaaSP version (SUSEConnect library)
+  - reports availability of new versions to velum via grains.
+
+  Furthermore if the system is registered against SMT/RMT:
+  - query RMT/SMT API if there is a new compatible version available for this CaaSP version
+  - reports if an admin needs to mirror the new version channel completely first (out of sync mirror)
+* get the link to the release notes
+
+also see section "Systems Registered against RMT/SMT"
+
+#### User Interface
+
+The user interface of velum needs to be adapted minimally.
+
+* [indirect] display of update content through [a link to the] release notes
+* display of update history via velum (optional)
+* button to trigger the maintenance updates (what we currently have)
+* button to switch to the next channel in velum
+
+Velum needs to know when an update is available, by looking at a special grain, that indicates
+the availability of an update or a new version. This grain is set by the services that run on
+each worker.
+
+#### Salt back-end
+
+##### Triggering updates/migrations
+
+We will continue to use `transactional-update` to trigger the updates and migrations.
+Salt will be used to call `transactional-update [dup|migrate]`
+
+We can reuse the update orchestration with an extra option to call `transactional-update migrate`.
+
+`transactional-update` will report the result back to velum via grains.
+
```

Comment:
I think we have to extend this section to make it clear that, at least in the first iteration, the maintenance updates are going to be handle in the current way. Meaning, `transactional-update` is run by a system timer, maintenance updates are going to be downloaded and installed automatically, once the new snapshot is ready a grain will be set. Velum will notice it and orchestrate the reboot after the user presses a button inside of velum's UI.

Migration updates are going to be handled differently because, as soon as the user presses the migrate button, salt will invoke the migration command and wait for its completion.
I think we should also incorporate the detailed steps @thkukuk provided inside of [this comment](https://github.com/kubic-project/rfc/pull/6#issuecomment-411997182) to ensure they are not lost.